src/debug/daccess/request.cpp

```diff
@@ -3226,8 +3225,22 @@ ClrDataAccess::GetDomainLocalModuleDataFromAppDomain(CLRDATA_ADDRESS appDomainAd
     pLocalModuleData->ModuleID = moduleID;
 
     AppDomain *pAppDomain = PTR_AppDomain(TO_TADDR(appDomainAddr));
+
     ModuleIndex index = Module::IDToIndex(moduleID);
-    DomainLocalModule* pLocalModule = pAppDomain->GetDomainLocalBlock()->GetModuleSlot(index);
+
+    DomainLocalModule* pLocalModule = NULL;
+
+    AppDomain::AssemblyIterator i = pAppDomain->IterateAssembliesEx(
+        (AssemblyIterationFlags)(kIncludeLoading | kIncludeLoaded | kIncludeExecution));
+    CollectibleAssemblyHolder<DomainAssembly *> pDomainAssembly;
+    while (i.Next(pDomainAssembly.This()))
+    {
+        if (pDomainAssembly->GetModule()->GetModuleIndex() == index)
+        {
+            pLocalModule = pDomainAssembly->GetModule()->GetDomainLocalModule();
```

Comment:
I do not think that this will ever find anything. Should the whole method just return `E_INVALIDARG`?

Reply:
Hmm, I was under the impression we still had Module indices. (I'm not sure why we would still have them, but I thought they were there.) In that case, I believe we could still find these things. I'll do some investigation.

Reply:
The module index is still used in `JIT_GetSharedNonGCThreadStaticBase` and `JIT_GetSharedGCThreadStaticBase` to find ThreadLocalModule using `ThreadStatics::GetTLMIfExists(index)`

Reply:
We do have module indexes, but we do not have module IDs. The code the calls `Module::IDToIndex(moduleID)` - like this method - should be unreachable.

Reply:
Sigh, right. I was ripping out these very bits of logic, and somehow missed this. Sigh, Well, I'll pull a bit more of those details out, as it was still confusing to me.

Reply:
And yes, since the moduleid here is an int, it will never match anything. E_INVALIDARG is correct.